dropwizard-db/src/main/java/io/dropwizard/db/DataSourceFactory.java

```diff
@@ -769,6 +813,12 @@ public ManagedDataSource build(MetricRegistry metricRegistry, String name) {
         poolConfig.setUrl(url);
         poolConfig.setUsername(user);
         poolConfig.setPassword(user != null && password == null ? "" : password);
+        poolConfig.setRemoveAbandoned(removeAbandoned);
+
+        if (getRemoveAbandonedTimeout().isPresent()) {
```

Comment:
Maybe we could set the timeout unconditionally? We already set the default value the same as in the pool configuration, if the user didn't specify the property.